cpp/cugraph_module/algorithms/betweenness_centrality.cu

```diff
@@ -0,0 +1,113 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mg_cugraph_utility.hpp"
+
+namespace {
+// TODO: Check Betweenness instances
+using vertex_t = int32_t;
+using edge_t = int32_t;
+using weight_t = double;
+using result_t = double;
+
+constexpr char const *kProcedureBetweenness = "get";
+
+constexpr char const *kArgumentNormalized = "normalized";
+constexpr char const *kArgumentDirected = "directed";
+constexpr char const *kArgumentWeightProperty = "weight";
+
+constexpr char const *kResultFieldNode = "node";
+constexpr char const *kResultFieldBetweennessCentrality = "betweenness_centrality";
+
+const double kDefaultWeight = 1.0;
+
+void InsertBetweennessRecord(mgp_graph *graph, mgp_result *result, mgp_memory *memory, const std::uint64_t node_id,
+                             double rank) {
+  auto *record = mgp::result_new_record(result);
+  mg_utility::InsertNodeValueResult(graph, record, kResultFieldNode, node_id, memory);
+  mg_utility::InsertDoubleValueResult(record, kResultFieldBetweennessCentrality, rank, memory);
+}
+
+void BetweennessProc(mgp_list *args, mgp_graph *graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    auto normalized = mgp::value_get_bool(mgp::list_at(args, 0));
+    auto directed = mgp::value_get_bool(mgp::list_at(args, 1));
+    auto weight_property = mgp::value_get_string(mgp::list_at(args, 2));
+
+    raft::handle_t handle{};
+    auto stream = handle.get_stream();
+
+    auto mg_graph = mg_utility::GetWeightedGraphView(
+        graph, result, memory, directed ? mg_graph::GraphType::kDirectedGraph : mg_graph::GraphType::kUndirectedGraph,
+        weight_property, kDefaultWeight);
+    if (mg_graph->Empty()) return;
+
+    auto n_vertices = mg_graph.get()->Nodes().size();
+    // IMPORTANT: Betweenness centrality cuGraph algorithm works only on legacy code
+    auto cu_graph_ptr =
+        mg_cugraph::CreateCugraphLegacyFromMemgraph<vertex_t, edge_t, weight_t>(*mg_graph.get(), handle);
+    auto cu_graph_view = cu_graph_ptr->view();
+    cu_graph_view.prop.directed = directed;
+
+    rmm::device_uvector<result_t> betweenness_result(n_vertices, stream);
+    // TODO: Add weights to the betweenness centrality algorithm
+    cugraph::betweenness_centrality<vertex_t, edge_t, weight_t>(handle, cu_graph_view, betweenness_result.data(),
+                                                                normalized, false, static_cast<weight_t *>(nullptr));
+
+    for (vertex_t node_id = 0; node_id < betweenness_result.size(); ++node_id) {
+      auto rank = betweenness_result.element(node_id, stream);
+      // There is an offset in the result
+      if (std::abs(rank) < 1e-20) rank = 0;
```

Comment:
What is this offset?

Reply:
Not sure but in the result we have something like 1e-20 which is clearly not the result we want.

Reply:
But isn't it that `ROUND` function should fix that, not us here?

Reply:
Yeah, you got the point